.github/workflows/publish.yaml

```diff
@@ -0,0 +1,35 @@
+on:
+  create:
+    tags:
+      - v*
+
+name: Publish Crate
+
+jobs:
+  publish:
+    name: Deploy a new version of Rspotify to crate.io
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - name: Release
+        env:
+          CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+        shell: bash
+        run: |
+          # Not sure if it's necessary
+          git config --global user.email "runner@gha.local"
+          git config --global user.name "Github Action"
+
+          chmod +x "${GITHUB_WORKSPACE}/tool/publish.sh"
+
+          # Publishing crates
+          "${GITHUB_WORKSPACE}/tool/publish.sh"
```

Comment:
this should be `tools`, not `tool` btw

Reply:
also, can you make `publish.sh` executable? `chmod u+x publish.sh` should work fine